### ISSUE TITLE
Support named child routes

### ISSUE THREAD

**Issue by hannesj** [4Catalyzer/found#55]
React-router provides a way to provide multiple components for a single route. It would be nice if found would match this functionality.

**Comment by taion** [comment]
See discussion in https://github.com/4Catalyzer/found/pull/67 – we have an idea for a better approach to do this.

**Comment by taion** [comment]
The API is going to look something like:

```js
<Route
  path="parent"
  Component={Parent}
>
  <Route
    path="foo"
    groups={{
      nav: (
        <Route path="*" Component={FooNav} />
      ),
      main: [
        <Route path="1" Component={Foo1} />, 
        <Route path="2" Component={Foo2}>
          <Route path="a" Component={Foo2a} />
        </Route>,
      ],
    }}
  />
  <Route
    path="bar"
    groups={{ ... }}
  />
</Route>
```

**Comment by alexhawkins** [comment]
@taion  Do you have an eta on this? My app heavily relies on named components that I adopted while using react-relay-router. I've since converted to found and found-relay.

**Comment by taion** [comment]
Maybe next week. I haven't had time to work on this yet.

**Comment by alexhawkins** [comment]
Thanks, much appreciated. Great work here.

**Comment by taion** [comment]
See https://github.com/4Catalyzer/found/pull/113.

**Comment by taion** [comment]
Released in v0.3.3.
